web/components/PetPostModal/PetPostModal.tsx

```diff
@@ -102,13 +103,78 @@ const data: FosterTypeData = {
 
 const FosterQuestionnaire = ({
   fosterType,
+  postId,
   isFormViewOpen,
   onFormViewClose,
 }: {
   fosterType: FosterType;
+  postId: ObjectId;
   isFormViewOpen: boolean;
   onFormViewClose: () => void;
 }) => {
+  const initialQuestionResponseData: QuestionResponseData = data[
+    fosterType
+  ].reduce(
+    (responses, question) =>
+      ({
+        ...responses,
+        [question.key]: "",
+      } as QuestionResponseData),
+    {}
+  );
+
+  const [fosterQuestionResponses, setfosterQuestionResponses] = useState(
+    initialQuestionResponseData
+  );
+  const mutation = trpc.post.offer.useMutation();
+  const toast = useToast();
+
+  function handleSubmission() {
+    let formIsValid = true;
+    for (const [key] of Object.entries(fosterQuestionResponses)) {
+      if (fosterQuestionResponses[key as keyof typeof QuestionKeys] === "") {
+        formIsValid = false;
+        break;
+      }
+    }
```

Comment:
You can `safeParse()` the zod schema you create to handle the validation. See `PostCreationModal` for an existing implementation of this